google/cloud/storage/object_test.cc

```diff
@@ -47,6 +51,167 @@ class ObjectTest : public ::testing::Test {
   ClientOptions client_options = ClientOptions(CreateInsecureCredentials());
 };
 
+TEST_F(ObjectTest, InsertObjectMedia) {
+  std::string text = R"""({
+      "name": "foo-bar/baz/1"
+})""";
+  auto expected = storage::ObjectMetadata::ParseFromJson(text);
+
+  EXPECT_CALL(*mock, InsertObjectMedia(_))
+      .WillOnce(Invoke(
+          [&expected](internal::InsertObjectMediaRequest const& request) {
+            EXPECT_EQ("foo-bar", request.bucket_name());
```

Comment:
If someone does something to break this when making a change and then runs the tests, what does the failure message look like here? I don't think something like `Failure: expected "foo-bar", got "foo-ba"` is terribly helpful :P

Here and throughout, making these values things like "test-bucket-name", "test-object-name", "test-object-contents", etc. would probably make these tests a bit easier to follow and debug.

... this is probably true for the rest of the tests as well, but we can make a TODO to fix them all at once.